src/absil/il.fs

```diff
@@ -3899,8 +3900,41 @@ let decodeILAttribData (ilg: ILGlobals) (ca: ILAttribute) =
       let v, sigptr = parseVal ty sigptr
       parseNamed ((nm, ty, isProp, v) :: acc) (n-1) sigptr
     let named = parseNamed [] (int nnamed) sigptr
-    fixedArgs, named     
-
+    fixedArgs, named
+    
+[<Literal>]
+let ByRefLikeMarker = "Types with embedded references are not supported in this version of your compiler."
+
+type ILTypeDef with
```

Comment:
We don't really use type augmentations like this in `il.fs`.  To keep with the style of the rest of `il.fs` I suggest just using a module function like this:

    let GetFilteredCustomAttributes (ilg: ILGlobals) (td: ILTypeDef) = ....

rather than an instance member.

Reply:
On second thoughts I think that this logic should probably be in `AttributeChecking.fs` - it is much more related to the F# compiler (`fsharp\*`) than the raw IL format (`absil\il.fs`).  Is there a specific reason we need to call `GetFilteredCustomAttributes` any other place except in `AttributeChecking.fs`?  Nowhere else checks for `ObsoleteAttribute`....

Reply:
We do, we need to filter it in NameResolution as well. That is what originally sparked this PR.

I agree that this is more related to the F# compiler rather than the raw IL format. However, we use the raw IL format types in the compiler directly when making decisions; therefore, they are an integral part of decision making. In a perfect world, the compiler wouldn't know about these raw IL format types and we would provide some sort of abstraction that internally would access these types; that abstraction would be the one responsible for the filtering. Since that is not the case, it's tricky who should be responsible for filtering; I still lean towards `ILTypeDef` being responsible based on this.